aten/src/ATen/native/mkldnn/Matmul.cpp

```diff
@@ -142,12 +207,16 @@ void mkldnn_matmul(
                 "mkldnn_matmul: mkldnn_matmul bf16 path needs a cpu with bf16 support");
   }
 #else
-  TORCH_CHECK(mkldnn_bf16_device_check(),
-    "mkldnn_matmul: mkldnn_matmul bf16 path needs the cpu support avx512bw, avx512vl and avx512dq, or AWS Graviton3");
-
-   TORCH_CHECK(mat1.scalar_type() == at::kBFloat16 &&
-               mat2.scalar_type() == at::kBFloat16 &&
-               result.scalar_type() == at::kBFloat16, "mkldnn_matmul:  only enabled for bf16 path");
+  TORCH_CHECK((mat1.scalar_type() == at::kBFloat16 || mat1.scalar_type() == at::kHalf) &&
+                 mat2.scalar_type() == mat1.scalar_type() &&
+                 result.scalar_type() == mat1.scalar_type(), "mkldnn_matmul:  only enabled for bf16 and fp16 path");
+  if (mat1.scalar_type() == at::kBFloat16) {
+    TORCH_CHECK(mkldnn_bf16_device_check(),
+    "mkldnn_matmul: mkldnn_matmul bf16 path needs the cpu support avx_ne_convert or avx512bw, avx512vl and avx512dq, or AWS Graviton3");
+  } else {
+    TORCH_CHECK(mkldnn_fp16_device_check(),
```

Comment:
Would be nice to check that it's called only for those two dtypes...
```suggestion
    TORCH_DEBUG_ASSERT(mat1.scalar_type() == at::kHalf); 
    TORCH_CHECK(mkldnn_fp16_device_check(),
```

Reply:
Revised as suggested